units/periodic_builds.go

```diff
@@ -148,20 +159,13 @@ func (j *periodicBuildJob) Run(ctx context.Context) {
 
 }
 
-func (j *periodicBuildJob) addVersion(ctx context.Context, definition model.PeriodicBuildDefinition) (string, error) {
+func (j *periodicBuildJob) addVersion(ctx context.Context, definition model.PeriodicBuildDefinition, mostRecentRevision string) (string, error) {
 	token, err := j.env.Settings().GetGithubOauthToken()
 	if err != nil {
 		return "", errors.Wrap(err, "getting GitHub OAuth token")
 	}
 
-	mostRecentVersion, err := model.VersionFindOne(model.VersionByMostRecentSystemRequester(j.ProjectID))
```

Comment:
VersionByMostRecentSystemRequester filters VersionRequester on 

 `VersionRequesterKey: bson.M{
        "$in": evergreen.SystemVersionRequesterTypes,
    }`
  and FindVersionByLastKnownGoodConfig switches it to  `evergreen.RepotrackerVersionRequester`. 
  
  Is that maybe part of why it was broken before?

Reply:
It sort of is! Really the issue is that we VersionByMostRecentSystemRequester doesn't consider if there's a stub version (i.e. a broken version), and at least for periodic builds, stub versions didn't have revisions, which is what caused the bug. There's also really no reason to consider all system requester types, since they use the configs from the commit versions (and also because that's really what we say the periodic build is testing, just the most recent commit).

I'm actually also realizing though that using FindVersionByLastKnownGoodConfig means we'd skip the most recent commit if it's got a broken config, which isn't quite what we promise, so I'll tweak this to just do a FindOne on the latest commit. (Verified that [commit stub versions will always have a revision](https://github.com/evergreen-ci/evergreen/blob/6c5e326eb294959ed5d9c99d11b3cb871d799a8c/repotracker/repotracker.go#L328)).